pylint/config/config_initialization.py

```diff
@@ -67,14 +67,17 @@ def _config_initialization(
         warnings.filterwarnings("ignore", category=DeprecationWarning)
         linter.load_config_file()
 
-    try:
-        with warnings.catch_warnings():
-            warnings.filterwarnings("ignore", category=DeprecationWarning)
-            linter.load_command_line_configuration(args_list)
-    except SystemExit as exc:
-        if exc.code == 2:  # bad options
-            exc.code = 32
-        raise
+    # pylint: disable-next=fixme
+    # TODO: Optparse: This has been disabled pre-maturely because it interferes with
+    # argparse option parsing for 'disable' and 'enable'.
+    # try:
+    #     with warnings.catch_warnings():
+    #         warnings.filterwarnings("ignore", category=DeprecationWarning)
+    #         linter.load_command_line_configuration(args_list)
+    # except SystemExit as exc:
+    #     if exc.code == 2:  # bad options
+    #         exc.code = 32
+    #     raise
```

Comment:
Yeah I wanted to remove this section all at once. I'm 99% sure we can do this now without any problems, but it will lead to a lot of coverage deprecations which I want to fix by adding deprecation tests.

I'll do a follow-up in which I clean `pylint/config/config_initialization.py` from any `optparse` related code and leave this comment for now as it is part of that.